File: 프로그래머스/레벨2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.js
function solution(numbers) {
    const result = [];

    numbers.forEach(num => {
        const bigNum = BigInt(num);
        
        // 첫번째 0은 1로, 뒤는 모두 0
        const biggerNum = bigNum + 1n;
        
        // 첫번째 0 이전은 0, 뒤는 모두 1
        const bitDiff = bigNum ^ (bigNum + 1n);
        
        // 우측 시프트 시에 첫번째 01을 10으로 바꾸는 효과
        const minNum = biggerNum + (bitDiff >> 2n);

        result.push(Number(minNum));
    });

    return result;
}

File: 프로그래머스/레벨2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.cpp
#include <vector>

using namespace std;

vector<long long> solution(vector<long long> numbers) {
    vector<long long> result;

    for (long long num : numbers) {
        // 첫번째 0은 1로, 뒤는 모두 0
        long long biggerNum = num + 1;
        
        // 첫번째 0 이전은 0, 뒤는 모두 1
        long long bitDiff = num ^ (num + 1);
        
        // 우측 시프트 시에 첫번째 01을 10으로 바꾸는 효과
        long long minNum = biggerNum + (bitDiff >> 2);

        result.push_back(minNum);
    }

    return result;
}